lib/types/addresstrie_test.go

```diff
@@ -0,0 +1,144 @@
+package types
+
+import (
+	"encoding/json"
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddressTrie(t *testing.T) {
+	t.Parallel()
+
+	at := NewAddressTrie(map[string]HostAddress{
+		// IPv4
+		"simple.io":              {IP: net.ParseIP("1.2.3.4")},
+		"simple.io:443":          {IP: net.ParseIP("1.2.3.4"), Port: 8443},
+		"with-port.io:80":        {IP: net.ParseIP("5.6.7.8"), Port: 80},
+		"with-port.io:443":       {IP: net.ParseIP("5.6.7.8"), Port: 8443},
+		"*.wildcard.io":          {IP: net.ParseIP("9.10.11.12")},
+		"specific.wildcard.io":   {IP: net.ParseIP("90.100.110.120")},
+		"*wildcard-2.io":         {IP: net.ParseIP("13.14.15.16")},
+		"specific.wildcard-2.io": {IP: net.ParseIP("130.140.150.160")},
+
+		// IPv6
+		"simple-ipv6.io":              {IP: net.ParseIP("aa::bb")},
+		"simple-ipv6.io:443":          {IP: net.ParseIP("aa::bb"), Port: 8443},
+		"with-port-ipv6.io:80":        {IP: net.ParseIP("cc::dd"), Port: 80},
+		"with-port-ipv6.io:443":       {IP: net.ParseIP("cc::dd"), Port: 8443},
+		"*.wildcard-ipv6.io":          {IP: net.ParseIP("ee::ff")},
+		"specific.wildcard-ipv6.io":   {IP: net.ParseIP("ee:11::ff")},
+		"*wildcard-2-ipv6.io":         {IP: net.ParseIP("aa::aa")},
+		"specific.wildcard-2-ipv6.io": {IP: net.ParseIP("a1::a1")},
+	})
+
+	tcs := []struct {
```

Comment:
I found this test table very hard to process and comprehend. The reason for it is that it depends on this big mapping that was set beforehand. Cognitively I can't keep all of it loaded in my head as I read the code, which means that later on, as a maintainer it would take me a lot of time to chase an issue using this test suite, or modify it.

I would recommend to either add a `name`, or `description` field to the test table, in which, for each case, the actual thing that is tested and its expected outcome are described. _e.g_ "using a wildcard over a domain with two subdomains defined should succeed with 2 results"

Or split and regroup the different categories of scenarios into dedicated sub-tests.

What do you think? 🙇🏻

Reply:
I agree with you, every case had a reason but I can not understand their reason at a first glance, it's a poorly written test 😞 . But I remember why I put it like this, it's very hard to categorize this _perfectly_ and that's why I passed that in the first place. Refactoring this 🤓 
-

Reply:
Thanks for doing this 🙇🏻

Reply:
My pleasure ❤️